pkg/frontend/validate.go

```diff
@@ -163,3 +166,43 @@ func validateAdminVMSize(vmSize string) error {
 	}
 	return nil
 }
+
+func (f *frontend) validateAndReturnInstallVersion(ctx context.Context, body *[]byte) (string, error) {
+	oc, err := installversion.FromExternalBytes(body)
+	if err != nil {
+		return "", api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidRequestContent, "", "The request content was invalid and could not be deserialized: %q.", err)
+	}
+
+	// If this request is from an older API or the user never specified
+	// the version to install we default to the InstallStream.Version
+	if len(oc.Properties.InstallVersion) == 0 {
```

Comment:
This is an optimization, because we can check if the request has specified an install-version or not and exit early. Avoiding having to list the DB contents for no reason.